### PR TITLE
Fail e2e tests only after saving screenshots

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
If e2e tests fail in CI, the screenshots are not saved.  This makes it hard to find out what failed in CI.

# Changes
- Fail CI only after the screenshots have been committed.  This way the screenshots are visible in the PR.

Note: All screenshots are already removed before each CI run and recommitted afresh, so committing screenshots of failed tests will not persist after the problem has been fixed.

# Tests
Here is an example of a screenshot of a failed test in a PR:
Screenshot from 2022-04-10 07-14-15

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -113,7 +113,11 @@ jobs:
           # wipe the screenshots, they will be re-committed
           rm -rf screenshots
           set -o pipefail
-          REDIRECT_TO_LEGACY=svelte npm run test | tee -a wdio.log
+          if REDIRECT_TO_LEGACY=svelte npm run test | tee -a wdio.log
```
Comment (nmattia) [inline .github/workflows/build.yml:116]: Do you really want to check the return code of `tee`?

```
$ if true | false; then echo yes; else echo no; fi
no
```

or maybe I'm missing something

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -113,7 +113,11 @@ jobs:
...
+          then echo Success
+          else echo Tests failed.  Proceeding but taking note.
+               touch .tests-failed
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply .github/workflows/build.yml:119]: Sure, I an use GH_ENV.

[the remark this answers is not in the document]
Comment (bitdivine) [reply .github/workflows/build.yml]: Just checking, is GITHUB_ENV supported by act?

[the remark this answers is not in the document]
Comment (bitdivine) [reply .github/workflows/build.yml]: Yes, it appears to be supported.

**Pull request closed without merging**